src/main/minio.js

```diff
@@ -1044,6 +1044,54 @@ export class Client {
     ], cb)
   }
 
+  // FIXME: doc
+  copyObject(bucketName, objectName, srcObject, conditions, cb) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
+    }
+    if (!isString(srcObject)) {
+      throw new TypeError('srcObject should be of type "string"')
+    }
+    if (srcObject == "") {
+      throw new errors.InvalidPrefixError(`Empty source prefix`)
+    }
+
+    if (conditions != null && !(conditions instanceof CopyConditions)) {
```

Comment:
Conditions can be null and optional,  we need to validate 'conditions' can be a 'function()' i.e a callback so this function can take such arguments

```
copyObject(bucketName, objectName, srcObject, cb)
```

```
copyObject(bucketName, objectName, srcObject, conds, cb)
``` 

is also valid.